deployer/hub.py

```diff
@@ -246,9 +246,8 @@ def get_generated_config(self, auth_provider: KeyProvider, secret_key):
         #
         # Allow explicilty ignoring auth0 setup
         if self.spec['auth0'].get('enabled', True):
-
             client = auth_provider.ensure_client(
-                self.spec['name'],
+                self.cluster.spec['name'],
```

Comment:
```suggestion
                f"{self.cluster.spec['name']}-{self.spec['name']}",
```

Reply:
Are there other places this change needs to happen?

Reply:
I don't understand your suggested change. When I look in the auth0 client list, I see no indication that the clients are named like this and therefore only created a single client called `jmte` rather than `jmte-staging` or similarly.

Do you suggest we start using `<clustername>-<hubname>` instead, and update our auth0 OAuth2 applications?

Reply:
> Are there other places this change needs to happen?

I think there are if we accept your commit suggestion as /hub/login has correctly used the correct client_id even though the /hub/logout didn't - which this was fixing.

Reply:
Just a side note: I think that if we go with `<clustername>` and have a connection per-cluster, then `redirectTo` should pick the right, hub-specific logout url, rather than the first one in the list:
https://github.com/2i2c-org/pilot-hubs/blob/27be2de984009da7733832ed188bc6e0e945328d/deployer/auth.py#L166

Reply:
@yuvipanda it is perhaps out of scope for this PR to also address this, but I figure this is a bug if many hubs use the same oauth2 application at least. Hmmm... But by going for a hub specific oauth2 application, this won't be an issue...

Okay! All good!